src/main/java/io/netty/handler/codec/compression/BrotliDecoder.java

```diff
@@ -54,58 +47,41 @@ public BrotliDecoder() {
 
   @Override
   protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) throws Exception {
-    /*
-       use in.alloc().buffer() instead of Unpooled.buffer() as best practice.
-       See: https://github.com/netty/netty/wiki/New-and-noteworthy-in-4.0#pooled-buffers
-    */
+    ByteBuf result = in.alloc().buffer();
     boolean success = true;
-    ByteBuf outBuffer = null;
-    try (ByteBufOutputStream output = new ByteBufOutputStream(in.alloc().buffer())) {
-      try (ByteBufInputStream bbin = new ByteBufInputStream(in)) {
-        try (BrotliInputStream brotliInputStream = new BrotliInputStream(bbin)) {
-          success = decompress(output, brotliInputStream);
-          if (!success) {
-            return;
-          }
-        }
-      }
-      outBuffer = output.buffer();
-      if (outBuffer.isReadable()) {
-        out.add(outBuffer);
-      } else {
-        log.warn("Could not read Brotli output.");
+    try (
+        ByteBufOutputStream bbout = new ByteBufOutputStream(result);
+        ByteBufInputStream bbin = new ByteBufInputStream(in);
+    ) {
+      decompress(bbin, bbout);
+      if (!result.isReadable()) {
         success = false;
+        return;
       }
+      out.add(result);
+    } catch (IOException ioe) {
+      // stream is corrupted or not ready (retriable)
+      // exit condition will be reached when Netty has nothing more to add to the buffer
+      // and this function is not able to decode any messages to "out"
+      success = false;
```

Comment:
looks like it just silently swallows the exception. isn't it?

Reply:
Hm, yeah, that's right. It is somewhat expected, but it doesn't look good still. I'll look for a way to log it without spamming, thanks.